src/hourly-weather.ts

```diff
@@ -320,15 +322,18 @@ export class HourlyWeatherCard extends LitElement {
     const forecastNotAvailable = !forecast || !forecast.length;
 
     if (numSegments < 1) {
+      if (pending) return;
       // REMARK: Ok, so I'm re-using a localized string here. Probably not the best, but it avoids repeating for no good reason
       return await this._showError(this.localize('errors.offset_must_be_positive_int', 'offset', 'num_segments'));
     }
 
     if (offset < 0) {
+      if (pending) return;
       return await this._showError(this.localize('errors.offset_must_be_positive_int'));
     }
 
     if (!forecastNotAvailable && numSegments > (forecast.length - offset)) {
+      if (pending) return;
```

Comment:
No need to block this either - in the case that the data is not yet loaded, `forecastNotAvailable` would be true, so this block wouldn't run anyway.

Reply:
I think we need it -- as it might happen that forecast exists as an attribute with (e.g. 6 segments), yet we are looking for more (e.g. 24) segments from the subscription.